sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/BeamSqlExpression.java

```diff
@@ -51,8 +51,8 @@ public SqlTypeName opType(int idx) {
   }
 
   public <T> T opValueEvaluated(
-      int idx, Row row, BoundedWindow window, BeamSqlExpressionEnvironment env) {
-    return (T) op(idx).evaluate(row, window, env).getValue();
+      Class<T> clazz, int idx, Row row, BoundedWindow window, BeamSqlExpressionEnvironment env) {
```

Comment:
I'm not a fan of this solution; it still seems like a misuse of the type system. In other places, I've removed the type parameter and returned an `Object` and let the callee cast it. Thoughts?

Reply:
Yeah honestly not a huge fan of this either, your solution seems a bit cleaner and more explicit. Let me try it out and see how it goes and I'll be able to defend a solution better.

Reply:
+1  just let callee cast it. Otherwise, LGTM.